utilities/write_batch_with_index/write_batch_with_index_test.cc

```diff
@@ -1814,6 +1812,103 @@ TEST_P(WriteBatchWithIndexTest, NewIteratorWithBasePrepareValue) {
   }
 }
 
+TEST_P(WriteBatchWithIndexTest, NewIteratorWithBaseAllowUnpreparedValue) {
+  ColumnFamilyHandleImplDummy cf1(1, BytewiseComparator());
+  KVMap map{{"a", "aa"}, {"c", "cc"}, {"e", "ee"}};
+
+  ASSERT_OK(batch_->Put(&cf1, "c", "cc1"));
+
+  ReadOptions read_options = read_opts_;
+  read_options.allow_unprepared_value = true;
+
+  {
+    std::unique_ptr<Iterator> iter(batch_->NewIteratorWithBase(
+        &cf1, new KVIter(&map, /* allow_unprepared_value */ true),
+        &read_options));
+
+    iter->SeekToFirst();
+    ASSERT_TRUE(iter->Valid());
+    ASSERT_OK(iter->status());
+    ASSERT_EQ(iter->key(), "a");
+    ASSERT_TRUE(iter->value().empty());
+    ASSERT_TRUE(iter->PrepareValue());
+    ASSERT_EQ(iter->value(), "aa");
+
+    iter->Next();
+    ASSERT_TRUE(iter->Valid());
+    ASSERT_OK(iter->status());
+    ASSERT_EQ(iter->key(), "c");
+    // This key is served out of the delta iterator so this PrepareValue() is a
+    // no-op
```

Comment:
nit: check `value()` to be non-empty before `PrepareValue()` too?

Reply:
For this specific key, `value()` is served out of the delta iterator (that is, the write batch), so it's non-empty even before the call to `PrepareValue` (so that call is a no-op).

Reply:
Oh sorry, I see you wrote *non*-empty :) Yes that makes sense.